Table View Panel

timio23 opened the pull request.

ComfortablyCoding reviewed (changes requested):
Overall looks good!

### Feedback
1. I am unable to select the `ID` field for sorting for this panel. Alternative panels (e.g. List) allow it. Was there some reasoning behind this? <img width="797" alt="Screenshot 2024-08-27 at 9 51 01 AM" src="https://github.com/user-attachments/assets/aa94a616-d95d-4d24-9c36-b819da5e3e85">
2. I would recommend not to have sort field as required, it should default to id.
3. I am unable to select core collections for the `Collection` input
4. See other review comments.

ComfortablyCoding commented on packages/table-view-panel/src/panel.vue:
```diff
@@ -0,0 +1,333 @@
+		}
+
+		// On load
+		fetchData({
```
We should probably trigger this within the `onMounted` hook

timio23 commented:
Thanks @ComfortablyCoding for your feedback. I've updated the panel and requested an another review. Regarding your comments:

1. IDs are usually GUID now which make sorting on ID unpredictable. Saying that I recognize that we should still give people the option so I've included the ID in the sort options.
2. Sort field is now optional and defaults to the primary key field
3. This was an oversight by me. I've enabled core collections for selection.
4. I've updated the panel with these suggestion.

I noticed a bug with the relational field after saving changes to row so I added a fix and some better error handling.

ComfortablyCoding reviewed (changes requested):
Thanks @timio23 ! The original feedback looks to be addressed.

While re-testing I have come across the following (final) 2 issues:
1. Sorting seems to be broken for `Disable Sorting` for `ID` field specifically, was this the unpredictable behaviour you referred to?

https://github.com/user-attachments/assets/aeca5ab8-5b0b-4240-a152-c8cdb20dfc3a

2. If a user does not have `read` permissions an unknown error is shown in the display. The expectation is a blank panel with no error will be shown.
<img width="766" alt="Screenshot 2024-08-29 at 10 54 04 PM" src="https://github.com/user-attachments/assets/45085dae-11e3-4562-89e4-5cb4dd3584f5">

timio23 commented:
Thanks @ComfortablyCoding, I've addressed the 2 points regarding sorting and permissions:
1. The sorting confusion is now fixe. I set the sorting constant to empty when "disable sort" is chosen (defaulting to the primary key).
2. The drawer-item was causing the reported issue. Adding the permissions check on the drawer-item resolved this which now shows my "No Items" warning. To show a blank panel as you expected, I had to move a few things around and I felt it still needed a notification when they don't have permissions or there are no items so I changed the v-info to v-notify in these situations. See what you think.

timio23 commented:
@ComfortablyCoding, those recommended changes have been made.

2. I've been testing with 8 fields which causes the "No Items" text to be off screen so I added the fixed notice to always be in view. To get the best of both, I have hidden the table's notice and changed the fixed notice to look the same and positioned it in the middle. The forbidden notice is by iteself.
3. Add to package.json

timio23 commented:
https://github.com/user-attachments/assets/4839cc6f-b8a2-4661-b1b1-4c5c01cbb76f

ComfortablyCoding commented:
> @ComfortablyCoding, those recommended changes have been made.
> 
> 2. I've been testing with 8 fields which causes the "No Items" text to be off screen so I added the fixed notice to always be in view. To get the best of both, I have hidden the table's notice and changed the fixed notice to look the same and positioned it in the middle. The forbidden notice is by iteself.
> 3. Add to package.json

Thanks @timio23, looks great. Any chance we can get the permission notice to be at the top? It might not be viewable/obvious as to why no data is loading for tall tables otherwise.

#### Current Notice Placement
<img width="638" alt="Screenshot 2024-08-30 at 11 48 35 AM" src="https://github.com/user-attachments/assets/9055c3e2-ee65-4b55-b0fd-92c4f15a35ab">

timio23 commented:
@ComfortablyCoding, would you prefer v-info:

<img width="906" alt="image" src="https://github.com/user-attachments/assets/b1d73ae3-41e8-4953-9c05-ebcf10b08b36">

ComfortablyCoding commented:
> @ComfortablyCoding, would you prefer v-info:
> 
> <img alt="image" width="906" src="https://private-user-images.githubusercontent.com/124673267/363243583-b1d73ae3-41e8-4953-9c05-ebcf10b08b36.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjUwMzM3MjIsIm5iZiI6MTcyNTAzMzQyMiwicGF0aCI6Ii8xMjQ2NzMyNjcvMzYzMjQzNTgzLWIxZDczYWUzLTQxZTgtNDk1My05YzA1LWViY2YxMGIwOGIzNi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwODMwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDgzMFQxNTU3MDJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT00MjExMjExYWUzMzQ1MzVjZTc5YmRkN2JlYjQ3YzIxODBhZjM5YThhOTM0MDYxMzk0M2QzM2IwY2Q5NTgzYTg5JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.Ug2_cgqZgx8P2Tw6RC5vGnsFqi57eTO8rS42TD1f8yo">

Yes

timio23 commented:
@ComfortablyCoding that's sorted now

ComfortablyCoding reviewed (approved):
LGTM 🚀 , thanks @timio23
